### PR TITLE
Resolve #152 and #125

### LINKED ISSUE #125: RestClient - MD5 not getting calculated
https://github.com/mgomes/api_auth/blob/5a1bafe7581b29188cc3487325bcbbdd0237764d/lib/api_auth/request_drivers/rest_client.rb#L32

```
def populate_content_md5
   return unless [:post, :put].include?(@request.method)
   @request.headers['Content-MD5'] = calculated_md5
   save_headers
end
```

In the above line `@request.method` returns a string. But the code is checking for a symbol. Because of this, when signing the request, the MD5 does not get added to the header at all.

Is this a bug or am I missing something here?

I am using api-auth version 2.0.1 and rest-client version 2.0.0 .

### LINKED ISSUE #152: POST requests is not authentic
I have a two requests get and post

```
request = RestClient::Request.new(
  url: "http://localhost:3000/api/v1/accounts",
  method: :get
)

signed_request = ApiAuth.sign!(request, access_id, secret_key)

res = signed_request.execute
```
GET request have a good response and "authentic" is true

```
request = RestClient::Request.new(
  url: "http://localhost:3000/api/v1/accounts",
  method: :post
)

signed_request = ApiAuth.sign!(request, access_id, secret_key)

res = signed_request.execute
```

POST request has "authentic" false

How to fix post requests?

### PR DESCRIPTION
**Pull request by fwininger**
This PR resolves #152 and resolves #125.

@kjg or @mgomes can you consider that ?

### CODE REVIEW

**Comment by kjg** [comment]
Looks good. Thanks so much! Can you please split out the travis / gemfile changes from the code changes and fix the failing spec?

**Comment by fwininger** [comment]
@kjg done!

I just remove the failing test because, it's no more possible to do a empty uri with RestClient.
I try to fix that with Net::HTTP, but I have the same issue : 

```ruby
Net::HTTP::Get.new('')
> ArgumentError: HTTP request path is empty
```

**Comment by fwininger** [comment]
I can't fix rubocop without #160

File: spec/headers_spec.rb [hunk 1]
```diff
@@ -7,14 +7,6 @@
       subject(:headers) { described_class.new(request) }
       let(:uri) { '' }
 
-      context 'empty uri' do
```
Comment (kjg) [inline spec/headers_spec.rb]: I this is a probably important spec that describes an edge case

Comment (fwininger) [reply spec/headers_spec.rb]: I don't think that this case is essentiel because the next case is pretty the same thing. However, I agree to add a test for that if you found me a Gem that allow this. Net::HTTP rand RestClient v2 raise an exeception if you use a empty string.

Comment (kjg) [reply spec/headers_spec.rb]: You're right. This test was [always specific to RestClient](https://github.com/mgomes/api_auth/commit/6c957342f830a3f1ccc3a326bd87220fbe59e4a2#diff-e8b2590d477903fed0685b0f801c057dL307), and since RestClient no longer allows it, the test provides no value and only causes failure.

File: lib/api_auth/request_drivers/rest_client.rb [hunk 2]
```diff
@@ -29,13 +29,13 @@ def calculated_md5
       end
 
       def populate_content_md5
-        return unless %i[post put].include?(@request.method)
+        return unless %i[post put].include?(@request.method.to_sym)
```
Comment (kjg) [inline lib/api_auth/request_drivers/rest_client.rb]: Since RestClient returns a string now (and probably moving forward), maybe it would make sense to change the `%i[post put]` to `%w[post put]` and maybe have `@request.method.to_s` for backwards compatibility

Comment (fwininger) [reply lib/api_auth/request_drivers/rest_client.rb]: If you prefer, done !
